src/osd/PrimaryLogPG.cc

```diff
@@ -769,6 +790,20 @@ class PGLSParentFilter : public PGLSFilter {
                       bufferlist& outdata) override;
 };
 
+bool PGLSNameFilter::filter(const hobject_t &obj,
+                              bufferlist& indata, bufferlist& outdata)
+{
+  // match user regex
+  if (regex != "" && !regex.empty() && regex != "no_reg") {
```

Comment:
@branch-predictor I guess this check could be removed if I follow the comment from Jason.

> Don't use "magic" values -- if the regex isn't going to be used, don't use the filter.